test/PrintAsObjC/discardableresult.swift

```diff
@@ -0,0 +1,34 @@
+// Please keep this file in alphabetical order!
+
+// REQUIRES: objc_interop
+
+// RUN: rm -rf %t && mkdir -p %t
+
+// FIXME: BEGIN -enable-source-import hackaround
+// RUN:  %target-swift-frontend(mock-sdk: -sdk %S/../Inputs/clang-importer-sdk -I %t) -emit-module -o %t %S/../Inputs/clang-importer-sdk/swift-modules/ObjectiveC.swift
+// RUN:  %target-swift-frontend(mock-sdk: -sdk %S/../Inputs/clang-importer-sdk -I %t) -emit-module -o %t  %S/../Inputs/clang-importer-sdk/swift-modules/CoreGraphics.swift
+// RUN:  %target-swift-frontend(mock-sdk: -sdk %S/../Inputs/clang-importer-sdk -I %t) -emit-module -o %t  %S/../Inputs/clang-importer-sdk/swift-modules/Foundation.swift
+// FIXME: END -enable-source-import hackaround
+
+// RUN: %target-swift-frontend(mock-sdk: -sdk %S/../Inputs/clang-importer-sdk -I %t) -import-objc-header %S/../Inputs/empty.h -emit-module -o %t %s
+// RUN: %target-swift-frontend(mock-sdk: -sdk %S/../Inputs/clang-importer-sdk -I %t) -import-objc-header %S/../Inputs/empty.h -parse-as-library %t/discardableresult.swiftmodule -typecheck -emit-objc-header-path %t/discardableresult.h
+
+// RUN: %FileCheck %s < %t/discardableresult.h
+// RUN: %check-in-clang %t/discardableresult.h
+
+import Foundation
+
+// CHECK-LABEL: @interface Test : NSObject
+
+class Test : NSObject {
+
+// CHECK-NEXT: - (NSInteger)nonDiscardable:(NSInteger)x SWIFT_WARN_UNUSED_RESULT;
+  func nonDiscardable(_ x: Int) -> Int { return x }
+// CHECK-NEXT: - (NSInteger)discardable:(NSInteger)x;
+  @discardableResult func discardable(_ x: Int) -> Int { return x }
```

Comment:
Rather than add a whole new file, I would just stick this in with methods.swift.